pkg/build/api/v1beta1/conversion.go

```diff
@@ -264,26 +264,41 @@ func init() {
 				ref.Registry = ""
 				out.ImageTag = ref.String()
 			}
+			if (out.To != nil || len(in.DockerImageReference) > 0) && len(out.Tag) == 0 {
+				out.Tag = imageapi.DefaultImageTag
+			}
 			return nil
 		},
 		func(in *BuildOutput, out *newer.BuildOutput, s conversion.Scope) error {
 			if err := s.Convert(&in.To, &out.To, 0); err != nil {
 				return err
 			}
-			out.Tag = in.Tag
 			out.PushSecretName = in.PushSecretName
 			if len(in.DockerImageReference) > 0 {
 				out.DockerImageReference = in.DockerImageReference
+				out.Tag = in.Tag
```

Comment:
again, not sure why we're setting out.Tag if there's a DockerImageReference.  Tag is only relevant if in.To is not nil.

and if in.To is not nil, then DockerImageReference is ignored.